api/serializers.py

```diff
@@ -1843,46 +1842,48 @@ def get_user_countries_regions(user):
 
     @staticmethod
     def get_local_unit_global_validators(user) -> List[int]:
-        data = list(
-            Permission.objects.filter(codename__startswith="local_unit_global_validator", group__user=user).values_list(
-                "codename", flat=True
-            )
-        )
-        global_validators = []
-        for code in data:
-            type_id = code.split("_")[-1]
-            global_validators.append((int(type_id)))
-        return global_validators
+        permission_codenames = Permission.objects.filter(
+            codename__startswith="local_unit_global_validator", group__user=user
+        ).values_list("codename", flat=True)
+        global_validators = {int(code.split("_")[-1]) for code in permission_codenames}
+        return list(global_validators)
 
     @staticmethod
-    def get_local_unit_region_validators(user) -> dict[str, list[int]]:
-        data = Permission.objects.filter(
+    def get_local_unit_region_validators(user) -> list[dict]:
+        permission_codenames = Permission.objects.filter(
             codename__startswith="local_unit_region_validator",
             group__user=user,
         ).values_list("codename", flat=True)
-
-        region_validators = defaultdict(set)
-
-        for code in data:
+        region_validators_map = {}
+        for code in permission_codenames:
             parts = code.split("_")[-2:]
             if len(parts) == 2 and all(p.isdigit() for p in parts):
-                type_id, region_id = map(int, parts)
-                region_validators[region_id].add(type_id)
-
-        return {(region_id): sorted(types) for region_id, types in region_validators.items()}
+                local_unit_type_id = int(parts[0])
+                region_id = int(parts[1])
+                region_validators_map.setdefault(region_id, set()).add(local_unit_type_id)
```

Comment:
This expression is complex for the larger dataset.
as setdefault() evaluates its default argument (set()) every time, even if not used.


```suggestion
                region_validators_map[region_id].add(local_unit_type_id)
```